uint32 seconds
uint32 nanoseconds
